java/src/jmri/ThrottleListener.java

```diff
@@ -38,67 +63,113 @@
-
+    
     /**
-     * Get notification that a throttle request requires is in use by another
-     * device, and a "steal" may be required.
+     * Get notification that a throttle request is in use by another
+     * device, and a "steal", "share", or "steal/share" decision may be required.
      *
-     * @param address LocoAddress of the throttle that needs to be stolen.
+     * @param address The LocoAddress that needs the decision.
+     * @param question The question being asked, steal / cancel, share / cancel, steal / share / cancel
      */
-    public void notifyStealThrottleRequired(LocoAddress address);
+    public void notifyDecisionRequired(LocoAddress address, DecisionType question);
 
 }
 
 /*
  * @startuml jmri/doc-files/ThrottleListener-Sequence.png
- * participant ThrottleListener
- * participant ThrottleManager
- * participant Throttle
+ *  participant ThrottleListener
+ *  participant ThrottleManager
+ * 
+ *  group ThrottleListener wants ThrottleManager to make steal / share decisions
+ * 
+ *      note over ThrottleListener : AutoDispatcher\nAbstractAutomation
+ *      ThrottleListener-> ThrottleManager : requestThrottle(LocoAddress,ThrottleListener,<b>false</b>)
+ *      ThrottleListener-> ThrottleManager : requestThrottle(BasicRosterEntry,ThrottleListener,<b>false</b>)
+ *  end
+ *  
+ *  group ThrottleListener wants to make a custom steal / share decision
+ *      note over ThrottleListener : AddressPanel\nWiThrottle
+ *      ThrottleListener-> ThrottleManager : requestThrottle(LocoAddress,ThrottleListener,<b>true</b>)
+ *      ThrottleListener-> ThrottleManager : requestThrottle(BasicRosterEntry,ThrottleListener,<b>true</b>)
+ *  end
+ *  
+ *  group If Throttle Request Cannot Continue
+ *     ThrottleListener <-- ThrottleManager : return false 
+ *     deactivate ThrottleManager
+ *     note over ThrottleListener : Request ends here
+ *  end
+ *  deactivate ThrottleListener
+ * 
+ *  group If Throttle Request Can Continue
+ *     ThrottleListener <-- ThrottleManager : return true
+ *     note over ThrottleListener : Waits for callback to proceed.
+ *  end
+ *     
+ *  group If Throttle Object exists
+ *     
+ *      note over ThrottleManager : Sets BasicRosterEntry ( if present ) to Throttle
+ *      ThrottleListener <-- ThrottleManager: notifyThrottleFound(DccThrottle)
+ *      note over ThrottleListener : Throttle can now be controlled by ThrottleListener or a delegate.
+ *     
+ *  end
+ *     
+ *  group If Throttle Object does not exist
+ *     
+ *      note over ThrottleManager : Throttle Manager starts system specific actions to create a throttle.
+ *      note over ThrottleManager : If the throttle is available by normal means, no stealing or sharing,\nThrottle creation continues normally.
+ *      
+ *      == Optional: A steal or share decision is needed requestThrottle( .. , ThrottleListener, false) ==
  *
- * == Quick Failure Scenario ==
```

Comment:
Can we put the whole quick failure scenario back in please.  This is still a valid scenario.

Reply:
Resolved